pkg/cmd/factory/http.go

```diff
@@ -57,8 +58,31 @@ type configGetter interface {
 }
 
 // generic authenticated HTTP client for commands
-func NewHTTPClient(io *iostreams.IOStreams, cfg configGetter, appVersion string, setAccept bool) *http.Client {
+func NewHTTPClient(io *iostreams.IOStreams, cfg configGetter, appVersion string, setAccept bool) (*http.Client, error) {
 	var opts []api.ClientOption
+
+	// We need to check and potentially add the unix socket roundtripper option
+	// before adding any other options, since if we are going to use the unix
+	// socket transport, it needs to form the base of the transport chain
+	// represented by invocations of opts...
+	//
+	// Another approach might be to change the signature of api.NewHTTPClient to
+	// take an explicit base http.RoundTripper as its first parameter (it
+	// currently defaults internally to http.DefaultTransport), or add another
+	// variant like api.NewHTTPClientWithBaseRoundTripper. But, the only caller
+	// which would use that non-default behavior is right here, and it doesn't
+	// seem worth the cognitive overhead everywhere else just to serve this one
+	// use case.
+	unixSocket, err := cfg.Get("", "http_unix_socket")
```

Comment:
The test data suggests that this should be a per-hostname setting which makes sense. However, that makes implementing this more complex -- we don't know what host a user wants to use until we have a request in hand we can extract a hostname from.

You're welcome to propose a refactor that will make using the socket hostname aware; otherwise, we can discuss the trade-off of making this a global setting instead of a per-host one.

Reply:
That's a good point -- thanks for raising it. How would you feel about me changing things to make it a global setting? Making it per-host does seem ideal, but I suspect it is a non-trivial refactor to get us into that position first. @mislav mentioned some work may be ongoing to that effect, but I wasn't clear what it is or where it is tracked ([discussion](https://github.com/cli/cli/pull/3308#issuecomment-812679219)).

In my experience at least, having it on the global config is still valuable. In fact that's how I am using it, as the code here indicates, and it seems like having a global config to start and later adding per-host overrides would be possible and let us make partial progress now. Would you be open to an update in which I make the test code align with this, and support it only at the host level?

Reply:
I'm totally fine with this being global to start :+1:

Reply:
Thanks. I backed out the test assertions which suggested there was any meaning to this value on a host-level :)

One thing that I can't tell for sure is whether config entries are supposed to be specifically marked somehow as "config-only", "host-only", or "cascading", etc. I don't think there is, and IIUC the config schema itself doesn't enforce where a config can be found -- that is enforced by convention only. If I overlooked it though please let me know.

Reply:
No, there is no way to be explicit about those config categories, unfortunately -- it's convention and documentation right now.